tests/cpptests/test_cpp_iterator_index.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Redis Ltd. 2016 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "gtest/gtest.h"
+#include "iterator_util.h"
+
+#include "src/forward_index.h"
+#include "src/iterators/inverted_index_iterator.h"
+
+typedef enum IndexType {
+    INDEX_TYPE_TERM_FULL,
+    INDEX_TYPE_NUMERIC_FULL,
+    INDEX_TYPE_TERM,
+    INDEX_TYPE_NUMERIC,
+    INDEX_TYPE_GENERIC,
+} IndexType;
+
+class IndexIteratorTest : public ::testing::TestWithParam<IndexType> {
+protected:
+    static constexpr size_t n_docs = 1234;
```

Comment:
isn't there a configuration to adapt the block size for the testing?

Reply:
Also, aren't we missing combinations of Seekers and Decoders being used? And multi option?